### ISSUE TITLE
xyTipsOn for LineChart

### ISSUE THREAD

**Issue by kum-deepak** [dc-js/dc.js#1724]
The document (http://dc-js.github.io/dc.js/docs/html/LineChart.html#xyTipsOn__anchor) suggests that `xyTipsOn` has no effect if brush in on. The document also suggests that it is `boolean`.

However, the code supports a special value 'always'  (https://github.com/dc-js/dc.js/blob/develop/src/charts/line-chart.js#L353).

@gordonwoodhull is it worth supporting that special value?

**Comment by gordonwoodhull** [comment]
Yes, it is. As with almost every feature, it is there because people use it.

I agree this is hacky interface for the feature, and if I were to do it again, I probably wouldn't do it this way.

The discussion is in #1152

**Comment by kum-deepak** [comment]
What is the way you would have done if doing afresh:

- three strings - 'always', 'never', 'default' (or, 'smart')
- Typescript Enum (which gets exposed as a class)
- Any other

We can decide and then implement it in other similar situations as well. We will maintain backward compatibility in compat layer.
